docs/Chipyard-Basics/Debugging-RTL.rst

```diff
@@ -0,0 +1,85 @@
+Debugging RTL
+======================
+
+While the packaged Chipyard configs and RTL have been tested to work,
+users will typically want to build custom chips by adding their own
+IP, or by modifying existing Chisel generators. Such changes might introduce
+bugs. This section aims to run through a typical debugging flow
+using Chipyard. We assume the user has a custom SoC configuration,
+and is trying to verify functionality by running some software test.
+
+Waveforms
+---------------------------
+
+The default SW simulators do not dump waveforms during execution. To build
+simulators with wave dump capabilities use must use the ``debug`` make target.
+For example:
+
+.. code-block:: shell
+
+   make CONFIG=CustomConfig debug
+
+The ``run-binary-debug`` rule will also automatically build a simulator,
+run it on a custom binary, and generate a waveform. For example, to run a
+test on ``helloworld.riscv``, use
+
+.. code-block:: shell
+
+   make CONFIG=CustomConfig run-binary-debug BINARY=helloworld.riscv
+
+VCS and Verilator also support many additional flags. For example, specifying
+the ``+vpdfilesize`` flag in VCS will treat the output file as a circular
+buffer, saving disk space for long-running simulations. Refer to the VCS
+and Verilator manuals for more information You may use the ``SIM_FLAGS``
+make variable to set additional simulator flags:
+
+.. code-block:: shell
+
+   make CONFIG=CustomConfig run-binary-debug BINARY=linux.riscv SIM_FLAGS=+vpdfilesize=1024
+
+Print Output
+---------------------------
+
+Both Rocket and BOOM can be configured with varying levels of print output.
+For information see the Rocket core source code, or the BOOM `documentation 
+<https://docs.boom-core.org/en/latest/>`__ .website. In addition, developers
```

Comment:
can you point to a specific place in the boom docs?

Reply:
I think we shouldn't point to a specific place here since subprojects are more likely to change their docs more rapidly (in theory)